llvm/lib/Transforms/Scalar/GVNHoist.cpp

```diff
@@ -996,8 +984,8 @@ unsigned GVNHoist::rauw(const SmallVecInsn &Candidates, Instruction *Repl,
         MSSAUpdater->removeMemoryAccess(OldMA);
       }
 
+      combineMetadataForCSE(Repl, I, true);
```

Comment:
Moving combineMetadataForCSE above andIRFlags is intentional for two reasons.
1. In future if andIRFlags does something to manipulate metadata(unlikely), calling it later makes this pass more resilient to change.
2. GVNSink takes a similar approach. https://github.com/llvm/llvm-project/blob/main/llvm/lib/Transforms/Scalar/GVNSink.cpp#L920